AppConnectSwift/CaptureImageViewController.swift

```diff
@@ -11,26 +11,68 @@ import UIKit
 class CaptureImageViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
     
     @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var saveImage: UIBarButtonItem!
+    
     var imagePicker = UIImagePickerController()
     var image = UIImage()
+    var userID : Int64!
+    var data : NSData!
+    var collectedSubjects: [MDSubject]!
+    var datastore = MDDatastoreFactory.create()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         takeOrSelectPicture(true)
         imagePicker.allowsEditing = false
         imagePicker.delegate = self
+        self.saveImage.enabled = false
+        self.loadSubjects()
     }
     
     override func viewDidAppear(animated: Bool) {
         super.viewDidAppear(animated)
         self.navigationItem.title = "Capture Image"
     }
     
+    func loadSubjects() {
+        var bgQueue : NSOperationQueue! = NSOperationQueue()
+        bgQueue.addOperationWithBlock() {
+            let clientFactory = MDClientFactory.sharedInstance()
+            let client = clientFactory.clientOfType(MDClientType.Network);
+            let user = self.datastore.userWithID(Int64(self.userID))
+            var loadedSubjectsAndErrors : [AnyObject] = []
+            
+            // Start an asynchronous task to load the subjects for the user logged in
+            client.loadSubjectsForUser(user) { (subjects: [AnyObject]!, error: NSError!) -> Void in
+                // Check all subjects loaded
+                if error != nil {
+                    loadedSubjectsAndErrors.append(error)
+                    if subjects != nil {
```

Comment:
Will there ever be a case where both `error` and `subjects` are not nil? What does the logic within the `error != nil` conditional do?